docs/proposals/multipart_trasnfer_mode.md

```diff
@@ -0,0 +1,296 @@
+* Just like `basic` transfers, if the file fully exists and is committed to storage, no `actions` will be provided
+  in the reply and the upload can simply be skipped
+* If a `verify` action is provided, calling it is required and not optional. In some cases, this endpoint may be used
+  to finalize the upload.
+* An empty or missing list of `parts` with a `verify` action may mean all parts have been uploaded but `verify` still
+  needs to be called by the client.
+* Authentication and authorization behave just like with the `basic` protocol.
+
+## Action Objects
+Each one of the `parts` and the `verify` and `abort` actions contain instructions for the client on how to send a
+request performing a the particular action. These are similar to `basic` transfer adapter `actions` but may include
+some common, as well as action specific additional parameters.
+
+All actions allow `href`, `header` and `expires_in` parameters just like `basic` transfer actions.
+
+### `parts` actions
+Each `parts` action should include the `pos` and `size` attributes, in addition to the attributes specified
+above:
+* `pos` indicate the position in bytes within the file in which the part should begin. If not specified, `0` (that is the
+beginning of the file) is assumed.
+* `size` is the size of the part in bytes. If `size` is omitted, default to read until the end of file.
+* If both `pos` and `size` are omitted, the action is expected to be a single-part upload of the entire file
+
+In adition, `parts` actions may include the following parameters:
+* `method`, with `PUT` as the default method if none is specified. This allows customizing the HTTP method used when uploading
+object parts.
+* `want_digest` to specify an expected HTTP `Digest` header, as described below.
+
+### `verify` action
+The `verify` action is similar to `basic` transfer mode `verify`, with the following additional parameters:
+* `params` - an object with additional parameters to send to the server as part of the `POST` JSON body.
+  These parameters are to be sent as provided back to the server.
+
+### `abort` action
+The `abort` action may include the `params` and `method` attributes, exactly as specified above.
+
+## Batch Request / Response Examples
+
+### Upload Batch Request
+The following is a ~10mb file upload request:
+```json
+{
+  "transfers": ["multipart", "basic"],
+  "operation": "upload",
+  "objects": [
+    {
+      "oid": "20492a4d0d84f8beb1767f6616229f85d44c2827b64bdbfb260ee12fa1109e0e",
+      "size": 10000000
+    }
+  ]
+}
+```
+
+### Upload Batch Response
+The following is a response for the same request, given an imaginary storage backend:
+
+```json
+{
+  "transfer": "multipart",
+  "objects": [
+    {
+      "oid": "20492a4d0d84f8beb1767f6616229f85d44c2827b64bdbfb260ee12fa1109e0e",
+      "size": 10000000,
+      "actions": {
+        "parts": [
+          {
+            "href": "https://foo.cloud.com/storage/upload/20492a4d0d84?part=0",
```

Comment:
Can we use the RFC 6761 TLD `.example` here and below (e.g., `foo.cloud.example` and `lfs.mycompany.example`)?  That way we won't accidentally conflict with a real domain.

Reply:
Fixed, BTW `.example` TLD is RFC-2606, RFC-6761 AFAIK is `example.com` and friends :wink: